File: Leetcode/leetcode-0771-JewelsAndStones-[string]-[easy]/Solution.py
class Solution(object):
    def numJewelsInStones(self, jewels, stones):
        """
        :type jewels: str
        :type stones: str
        :rtype: int
        """
        is_jewels = dict({})
        for jewel in jewels:
            is_jewels.update({jewel: True})
        
        cnt = 0
        for stone in stones:
            if (is_jewels.get(stone)):
                cnt = cnt + 1
        
        return cnt

File: Leetcode/leetcode-0771-JewelsAndStones-[string]-[easy]/Solution.java
class Solution {
    public int numJewelsInStones(String jewels, String stones) {
        HashSet<Character> isJewel = new HashSet<>();
        for (char jewel : jewels.toCharArray()) {
            isJewel.add(jewel);
        }

        int cnt = 0;
        for (char stone : stones.toCharArray()) {
            if (isJewel.contains(stone)) {
                cnt++;
            }
        }

        return cnt;
    }
}